tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_resource_mappings.go

```diff
@@ -0,0 +1,94 @@
+package dataapigeneratorjson
+
+import (
+	"encoding/json"
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+)
+
+type ResourceMapping struct {
+	ResourceIdMapping             *[]ResourceIdMapping            `json:"ResourceIdMapping"`
+	DirectAssignmentFieldMappings *[]DirectAssignmentFieldMapping `json:"DirectAssignmentFieldMappings,omitempty"`
+	ModelToModelFieldMappings     *[]ModelToModelFieldMapping     `json:"ModelToModelFieldMappings,omitempty"`
+}
+
+type ResourceIdMapping struct {
+	MappingFunction string
+	SegmentName     string
+}
```

Comment:
I think this should look more like 

```suggestion
type ResourceIdMapping struct {
	SchemaFieldName    string
	SegmentName        string
	ParsedFromParentID bool
}
```

The `ParsedFromParentID` flag determines whether we use the mapping function `ToCommonIdSegmentNamed` or `ToResourceIdSegmentNamed` in the C# files output by the old `dataapigenerator`. In this case I don't think there's any need to differentiate via a function name, the presence of the flag should give us all the information we need.